src/contrib/cluster/Akka.Cluster.Tools/Singleton/ClusterSingletonManager.cs

```diff
@@ -696,53 +696,55 @@ private void InitializeFSM()
                     else
                         return GoTo(ClusterSingletonState.Younger).Using(new YoungerData(initialOldestState.Oldest));
                 }
-                else return null;
+
+                return null;
             });
 
             When(ClusterSingletonState.Younger, e =>
             {
-                DelayedMemberRemoved removed;
-                YoungerData youngerData;
-                if (e.FsmEvent is OldestChangedBuffer.OldestChanged && (youngerData = e.StateData as YoungerData) != null)
+                if (e.FsmEvent is OldestChangedBuffer.OldestChanged oldestChanged
+                    && e.StateData is YoungerData youngerData)
```

Comment:
@Aaronontheweb Yes we could, but there is some limitations (or bug) in C# implementation. For example, I could use `case patterns` and use a variable with the same name in more that one place
```C#
string obj = "apple";
switch (obj)
{
    case null:
        Console.WriteLine("not a string");
        break;
    case string s when s.Length > 10:
        Console.WriteLine($"long string: {s}");
        break;
    case string s:
        Console.WriteLine($"string: {s}");
        break;
}
```

But I could not do the same with `if-patterns`
```C#
string obj = "apple";
if (obj == null)
{
	Console.WriteLine("not a string");
}
else if (obj is string s && s.Length > 0)
{
	Console.WriteLine($"long string: {s}");
}
else if (obj is string s) // Compiler error
{
	Console.WriteLine($"string: {b}");
}
```